plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -689,6 +710,31 @@ public boolean isLimitGuaranteed(ConnectorSession session)
         return true;
     }
 
+    private static ColumnMapping charColumnMapping(int charLength)
+    {
+        if (charLength > CharType.MAX_LENGTH) {
+            return varcharColumnMapping(charLength);
```

Comment:
Will this still preserve the char semantics of padding values?

There should be a test.

Reply:
there is a test

Reply:
good catch.  it doesn't correctly pad on write path and during predicate pushdown (this may or may not be relevant, depending on remote's coercions, need checking)

it's pre-existing behavior though and applies to other connectors as well, see `io.trino.plugin.jdbc.StandardColumnMappings#defaultCharColumnMapping`

do you want to file an issue for this?

Reply:
https://github.com/trinodb/trino/issues/7152 - the title could be better though.